core/src/main/scala/kafka/server/OffsetCheckpoint.scala

```diff
@@ -31,39 +31,45 @@ class OffsetCheckpoint(val file: File) extends Logging {
 
   def write(offsets: Map[TopicAndPartition, Long]) {
     lock synchronized {
-      // write to temp file and then swap with the existing file
-      val temp = new File(file.getAbsolutePath + ".tmp")
-
-      val fileOutputStream = new FileOutputStream(temp)
-      val writer = new BufferedWriter(new OutputStreamWriter(fileOutputStream))
       try {
-        // write the current version
-        writer.write(0.toString)
-        writer.newLine()
-      
-        // write the number of entries
-        writer.write(offsets.size.toString)
-        writer.newLine()
+        // write to temp file and then swap with the existing file
+        val temp = new File(file.getAbsolutePath + ".tmp")
+
+        val fileOutputStream = new FileOutputStream(temp)
+        val writer = new BufferedWriter(new OutputStreamWriter(fileOutputStream))
+        try {
+          // write the current version
+          writer.write(0.toString)
+          writer.newLine()
 
-        // write the entries
-        offsets.foreach { case (topicPart, offset) =>
-          writer.write("%s %d %d".format(topicPart.topic, topicPart.partition, offset))
+          // write the number of entries
+          writer.write(offsets.size.toString)
           writer.newLine()
+
+          // write the entries
+          offsets.foreach { case (topicPart, offset) =>
+            writer.write("%s %d %d".format(topicPart.topic, topicPart.partition, offset))
+            writer.newLine()
+          }
+
+          // flush the buffer and then fsync the underlying file
+          writer.flush()
+          fileOutputStream.getFD().sync()
+        } finally {
```

Comment:
Yeah I know. But if adding the catch in this existing try/finally can fix the same problem this patch intends to address, why not doing this since it requires less code and solve the problem in a more fine-granular manner?